app/src/main/java/org/mozilla/focus/exceptions/ExceptionsListFragment.kt

```diff
@@ -121,16 +123,18 @@ open class ExceptionsListFragment : BaseSettingsLikeFragment(), CoroutineScope {
         }
 
         removeAllExceptions.setOnClickListener {
-            requireComponents.trackingProtectionUseCases.removeAllExceptions()
+            launch(coroutineContext) {
+                requireComponents.trackingProtectionUseCases.removeAllExceptions()
 
-            val exceptions = (exceptionList.adapter as DomainListAdapter).selection()
-            TelemetryWrapper.removeAllExceptionDomains(exceptions.count())
-
-            requireComponents.appStore.dispatch(
-                AppAction.NavigateUp(
-                    requireComponents.store.state.selectedTabId
+                val exceptions = (exceptionList.adapter as DomainListAdapter).selection()
+                TelemetryWrapper.removeAllExceptionDomains(exceptions.count())
+                delay(REMOVE_EXCEPTIONS_DELAY)
```

Comment:
I have mixed feelings about this but I didn't find a better solution without doing changes in AC(maybe that's the only way). I'm open to suggestions if you have 🙏

Reply:
No problem, I'm taking a look to what alternatives we have. I'll report back :)